src/containers/createRoom.js

```diff
@@ -9,35 +9,91 @@ export class createRoom extends Component {
     username: "",
   };
 
-  handleChange = (event) => {
+  handleChange(event) {
     this.setState({ [event.target.name]: event.target.value });
   };
 
-  handleSubmit = (event) => {
-    event.preventDefault();
+  getRooms() {
     const reqObj = {
       method: "POST",
       headers: {
         "Content-Type": "application/json",
         Accept: "application/json",
       },
-      body: JSON.stringify({ room: this.state }),
+      body: JSON.stringify({
+        room: this.state
+      }),
     };
-    fetch("http://localhost:3000/rooms", reqObj)
-      .then((resp) => resp.json())
-      .then((resp) => {
-        localStorage.setItem("token", resp.jwt);
-        this.props.setCreateRoom(
-          resp.user,
-          resp.room.room_name,
-          resp.room.host_id,
-          resp.room.host_name
-        );
-        this.props.history.push(`/room/${resp.room.id}`);
-      });
+
+    return fetch("http://localhost:3000/rooms", reqObj)
+      .then((resp) => resp.json());
```

Comment:
Depending on your platform, you may also have access to `async`/`await` syntax. It's an evolution on asynchornous code management that allows you to operate on promises with a more procedural style.

For example, this function could look like:

```
async getRooms() {
  let result = await fetch("http://localhost:3000/rooms", {
    method: "POST",
    headers: {
      "Content-Type": "application/json",
      Accept: "application/json",
    },
    body: JSON.stringify({
      room: this.state
    }),
  });

  return result.json();
}
```

The `async` keyword says this function is asynchronous and will wrap any return value in a `Promise`. The `await` keyword takes a promise and returns the `.then((value) => {})` value from it.

I can explain this in more detail, if you like.